# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: toolkit/components/credentialmanagement/nsICredentialChooserService.idl
# Commit: d07def1365a4
# Full Hash: d07def1365a49dc8ecbf5b6c10fdfbd5ec750932
# Author: pstanciu <pstanciu@mozilla.com>
# Date: 2025-06-11 18:38:51
# Regressor Bug: 1969017
# File Overlap Count: 2
# Description:
#   Revert "Bug 1969015, 1969017: apply code formatting via Lando" for causing WPT failures @lfedcm-identity.create-store-collect.tentative.sub.https.html
#   
#   This reverts commit 03d4474e360f28be0fb075379f4e4535bd34b252.
#   
#   Revert "Bug 1969017 - Refactor FedCM to use a managed actor rather than tagging onto WindowGlobalParent - r=anti-tracking-reviewers,emz"
# ==============================================================================

diff -r 4e25ddaadc86 -r d07def1365a4 toolkit/components/credentialmanagement/nsICredentialChooserService.idl
--- a/toolkit/components/credentialmanagement/nsICredentialChooserService.idl	Wed Jun 11 13:42:13 2025 +0000
+++ b/toolkit/components/credentialmanagement/nsICredentialChooserService.idl	Wed Jun 11 18:04:27 2025 +0300
@@ -14,6 +14,20 @@
 
 [scriptable, uuid(673ddc19-03e2-4b30-a868-06297e8fed89)]
 interface nsICredentialChooserService : nsISupports {
+  /**
+   * Show UI to allow the user to make a choice between credentials
+   *
+   * @param browsingContext The top-level browsing context of the context requesting a credential.
+   * @param credentials The options the user should choose from
+   * @param callback A callback to notify when the user makes a choice
+   */
+  void showCredentialChooser(in BrowsingContext browsingContext, in Array<jsval> credentials, in nsICredentialChosenCallback callback);
+  /**
+   * Dismiss the credential chooser UI
+   *
+   * @param browsingContext The top-level browsing context of the context requesting a credential.
+   */
+  void cancelCredentialChooser(in BrowsingContext browsingContext);
 
    /**
    * A service function to help any UI. Fetches and serializes images to